src/libraries/Common/src/System/Security/Cryptography/MLDsa.cs

```diff
@@ -546,15 +630,16 @@ public string ExportEncryptedPkcs8PrivateKeyPem(
             ReadOnlySpan<char> password,
             PbeParameters pbeParameters)
         {
+            ArgumentNullException.ThrowIfNull(pbeParameters);
+            PasswordBasedEncryption.ValidatePbeParameters(pbeParameters, password, ReadOnlySpan<byte>.Empty);
             ThrowIfDisposed();
 
-            // TODO: Validation on pbeParameters.
-
             AsnWriter writer = ExportEncryptedPkcs8PrivateKeyCore(password, pbeParameters);
 
             try
             {
-                return writer.Encode(static span => PemEncoding.WriteString(PemLabels.EncryptedPkcs8PrivateKey, span));
+                // Skip clear since the data is already encrypted.
+                return EncodeAsnWriterToPem(PemLabels.EncryptedPkcs8PrivateKey, writer, clear: false);
```

Comment:
You comment why clear can be false, and pass false as clear... but are still in a try/finally that calls Reset on the writer.

I think the try/finally can be removed.